src/haz3lcore/statics/TypBase.re

```diff
@@ -365,6 +397,13 @@ module rec Typ: {
     | _ => (Unknown(Internal), Unknown(Internal))
     };
 
+  let matched_forall = (ctx, ty) =>
+    switch (weak_head_normalize(ctx, ty)) {
+    | Forall(t, ty) => (t, ty)
+    | Unknown(SynSwitch) => ("matched_forall", Unknown(SynSwitch))
```

Comment:
this could be resolved if we added a TPat hole in the Typ representation -- don't need a fresh/dummy variable